### ISSUE TITLE
Unable to run UI from a base path that is NOT "/"

### ISSUE THREAD

**Issue by marcantoine-bibeau** [Apicurio/apicurio-registry#5396]
## Description

I would like to run ApiCurio UI on my kube cluster at a different path then root
By example: 
https://app.domain.net/somewhere/apiregistry

**Registry
**Version**:  3.0.2
**Persistence type**: posgress

### Environment
Kubernetes 1.31.1

### Steps to Reproduce

I tried some configuration but they are not loaded. Was probably working before UI split ?
  REGISTRY_CONTEXT_PATH: "/somewhere/apiregistry"
  REGISTRY_NAV_PREFIX_PATH: "/somewhere/apiregistry"

From the browser, when I reach `/explore` the index.html download  `/config.js` file generated by the create-config.cjs (from entrypoint.sh). The problem is that in my case, `/config.js` does not exists. In my case it should be `/somewhere/apiregistry/config.js`. BUT ideally, it should be relative like `./config.js`

### Expected vs Actual Behaviour

Use relative path for ui to be able fetch configuration script (and version.js) to be able to pass environment variable such as REGISTRY_CONTEXT_PATH and REGISTRY_NAV_PREFIX_PATH

### Logs

![image](https://github.com/user-attachments/assets/17af0360-6dc6-4337-8a08-8e37c8b152ed)

### Hint
I am far from a FrontEnd expert but I was able to make one of my simple React app and run it under a custom path by using 
`"homepage": "."` in package.json file

**Comment by EricWittmann** [comment]
Possible duplicate of:  https://github.com/Apicurio/apicurio-registry/issues/5366

**Comment by marcantoine-bibeau** [comment]
It is a duplicate, I agree, sorry, did not find it  when I searched.

Can of hard for me to share my Kubernetes configuration because there a LOT (I'm using Istio virtual service) ;)
 
As I was saying, I did a FE app and got exactly the same problem ;)
What I did:
- In package.json: Added "homepage": "."
- In index.html: I'm using ./config.js (instead of /config.js)
- After, the REGISTRY_CONTEXT_PATH and REGISTRY_NAV_PREFIX_PATH should doo the tricks

I'm very ignorant in FE dev but if it worked for my spa, it should works for your since I was using the same tech stack

**Comment by EricWittmann** [comment]
No worries, I was just hoping you had a simple config you could share to bootstrap reproducing this. :)

Also thanks for the bulleted notes - that's consistent with what I would expect is needed and very helpful!

**Comment by EricWittmann** [comment]
Reference that might help depending on desired config:

* https://stackoverflow.com/questions/45981794/add-context-path-for-an-app-on-nginx

**Comment by EricWittmann** [comment]
OK I have something to report on this topic.  :)  I've been working on this (done and working for Apicurio Studio, will apply the same changes to Registry UI tomorrow) and here's what I've found (in no particular order):

1. The base URL needs to be set in `vite.config.js` instead of adding it to `package.json` as mentioned above.  That's just the difference between using something like `create-react-app` vs `vite` (we use the latter).  Same effect though:  the `index.html` contains relative paths instead of absolute ones.
2. Changed the references to `config.js` and `version.js` to be relative (`./config.js` etc).
3. I added `<base href="/">` to the `index.html` file, and modify its value based on the `APICURIO_CONTEXT_PATH` environment variable.
4. Must configure `APICURIO_CONTEXT_PATH` to whatever you want the context path to be.

So @marcantoine-bibeau - you were really close on this.  Really the only difference is (3) above, which is needed so that deep linking into the application will work.  Without that, the main UI will load, but a browser reload on any page within the application will fail.  For example, with a context path of "registry-ui", you might load the UI and then navigate to the following page:

https://apicurio.my-company.com/registry-ui/explore/apicurio/apicurio-registry-api/versions

Loading that URL directly (or just refreshing the browser once navigated to it) will fail because, although the application's `index.html` file will be served, all of its `href`s will be relative paths, resulting in the browser trying to load assets at locations like this:

* https://apicurio.my-company.com/registry-ui/explore/apicurio/apicurio-registry-api/versions/config.js
* https://apicurio.my-company.com/registry-ui/explore/apicurio/apicurio-registry-api/versions/assets/index-pGNTyKN6.js

So the addition of `<base href="/registry-ui/">` will fix that, causing the browser to load those same relative paths from these locations:

* https://apicurio.my-company.com/registry-ui/config.js
* https://apicurio.my-company.com/registry-ui/assets/index-pGNTyKN6.js

Success!

When I open a PR for this work in Registry I will include a docker compose configuration that replicates the setup and can serve as an example of the minimal set of ENV variables to make everything work.

**Comment by EricWittmann** [comment]
This should be fixed in PR:

* https://github.com/Apicurio/apicurio-registry/pull/5445

**Comment by EricWittmann** [comment]
@marcantoine-bibeau Once this PR is merged, it should be possible for you to test it in your configuration by using the `latest-snapshot` container image tags for both the UI and API images.  I wouldn't say no to that extra testing prior to an actual upstream release.  :)

**Comment by EricWittmann** [comment]
Fixed in #5445
